Mathlib/Algebra/Field/Subfield/Basic.lean

```diff
@@ -444,6 +444,21 @@ theorem coe_rangeRestrictField (f : K →+* L) (x : K) : (f.rangeRestrictField x
 theorem rangeRestrictField_bijective (f : K →+* L) : Function.Bijective (rangeRestrictField f) :=
   (Equiv.ofInjective f f.injective).bijective
 
+/--
+`RingHom.rangeRestrictField` as a `RingEquiv`.
+-/
+noncomputable def rangeRestrictFieldEquiv (f : K →+* L) : K ≃+* f.fieldRange :=
+  RingEquiv.ofBijective f.rangeRestrictField f.rangeRestrictField_bijective
+
+@[simp]
+theorem coe_rangeRestrictFieldEquiv (f : K →+* L) (x : K) :
+    (f.rangeRestrictFieldEquiv x : L) = f x := rfl
+
+@[simp]
+theorem rangeRestrictFieldEquiv_apply_symm_apply (f : K →+* L) (x : f.fieldRange) :
+    f (f.rangeRestrictFieldEquiv.symm x) = x := by
+  rw [← coe_rangeRestrictFieldEquiv, RingEquiv.apply_symm_apply]
```

Comment:
Can these simp lemmas be auto-generated with `@[simps]` on the definition?

Reply:
Yes, `simps` can correctly generate the first lemma, but not `rangeRestrictFieldEquiv_apply_symm_apply`.